tests/test_ops/test_ops.py

```diff
@@ -1096,3 +1096,36 @@ def forward(self, x):
             3: 'w'
         }),
         save_dir=save_dir)
+
+
+@pytest.mark.parametrize('backend', [TEST_TENSORRT])
+def test_dot_product_attention(backend, save_dir=None):
+    backend.check_env()
+
+    B = 2
+    Nt = 4
+    Ns = 4
+    E = 2
+    query = torch.rand(B, Nt, E)
+    key = torch.rand(B, Ns, E)
+    value = torch.rand(B, Ns, E)
+    mask = torch.rand(B, Nt, Ns)
+
+    def wrapped_function(query, key, value, mask=None):
+        from torch.nn.functional import _scaled_dot_product_attention
+        return _scaled_dot_product_attention(query, key, value, mask)
+
+    wrapped_model = WrapFunction(wrapped_function).eval()
+
+    with RewriterContext(
+            Config({'backend_config': {
+                'type': backend.backend_name
+            }}),
+            backend=backend.backend_name,
+            opset=11):
+        backend.run_and_validate(
```

Comment:
This testcase seems not make sure the precision.

I think that we should construct a whole multiheadattention opr and compare the result with `torch.nn.MultiHeadAttention`.

Reply:
Done.
Precision will be tested inside [run_and_validate](https://github.com/open-mmlab/mmdeploy/blob/792c27b054b4a93f81d00e69a0d7f162b8fcafc0/tests/test_ops/utils.py#L158).